att_hci.go

```diff
@@ -1021,32 +1044,43 @@ func (a *att) handleWriteReq(handle, attrHandle uint16, data []byte) error {
 	return a.sendError(handle, attOpWriteReq, attrHandle, attErrorWriteNotPermitted)
 }
 
-func (a *att) clearResponse() {
-	a.responded = false
-	a.errored = false
-	a.lastErrorOpcode = 0
-	a.lastErrorHandle = 0
-	a.lastErrorCode = 0
-	a.value = []byte{}
+func (a *att) clearResponse(handle uint16) error {
+	cd, err := a.findConnectionData(handle)
+	if err != nil {
+		return err
+	}
+
+	cd.responded = false
+	cd.errored = false
+	cd.lastErrorOpcode = 0
+	cd.lastErrorHandle = 0
+	cd.lastErrorCode = 0
+	cd.value = []byte{}
+
+	return nil
 }
 
-func (a *att) waitUntilResponse() error {
+func (a *att) waitUntilResponse(handle uint16) error {
+	cd, err := a.findConnectionData(handle)
+	if err != nil {
+		return err
+	}
+
 	start := time.Now().UnixNano()
 	for {
 		if err := a.hci.poll(); err != nil {
 			return err
 		}
 
 		switch {
-		case a.responded:
+		case cd.responded:
 			return nil
 
+		case (time.Now().UnixNano()-start)/int64(time.Second) > defaultTimeoutSeconds:
```

Comment:
I know this is outside the scope of the PR, but is this equivalent code for the time math?

```
timeout := time.Now().Add(defaultTimeoutSeconds)
for {
    ...
    switch {
        case time.Now().After(timeout):
            return ErrATTTimeout
```

Reply:
I would guess it might be a little faster due to the integer part of the calculation? I also copied that from elsewhere, so perhaps @aykevl can elaborate on why (or why not?)

Reply:
Honestly no idea, you'd have to look at the source of `.Add` or do a benchmark to check.
In general, when in doubt (or without a benchmark) I recommend taking the simpler approach - in this case, `time.Now().Add`.

Reply:
I need to replace some of these sprinkled thru my code...